packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -66,10 +71,22 @@ contract FuelERC20GatewayV4 is
     bool public whitelistRequired;
     bytes32 public assetIssuerId;
 
-    mapping(address => uint256) internal _deposits;
-    mapping(address => uint256) internal _depositLimits;
-    mapping(address => uint256) internal _decimalsCache;
-    mapping(bytes32 => bool) internal _isBridge;
+    mapping(address => uint256) _deposits;
+    mapping(address => uint256) _depositLimits;
+    mapping(address => uint256) _decimalsCache;
+    mapping(bytes32 => bool) _isBridge;
+
+    /// @notice Amounts already withdrawn this period for each token.
+    mapping(address => uint256) public rateLimitDuration;
```

Comment:
let's the take the chance to implement https://github.com/FuelLabs/fuel-bridge/issues/173

until mainnet launch we won't care if the erc20 gateway breaks backwards compatibility in storage

Reply:
Ignore the above comment, we are finally going to go with the old storage